### PR TITLE
Refactor RPC methods with dependency injection

### PR DESCRIPTION
**Pull request by hamstu**
## Description

Further refactor and improve our RPC Methods. 

* Automatically `require()` all the ones in the `server/rpc` directory 
* Update `@bufferapp/buffer-rpc` with it's new dependency injection feature so methods are easier to write and test.

## Context & Notes

* All RPC Methods can now access `PublishAPI` and `parsers` by destructuring the last parameter to their method. See diff for examples of usage. For how this change works in the RPC Server see https://github.com/bufferapp/buffer-rpc/pull/10
* These changes make RPC methods more simple to write; [for an example see this diff](https://github.com/bufferapp/buffer-publish/pull/1251/files#diff-34cecff1daede82048d6ccb21b0bd193L1):
image

* This makes testing bit easier as we can inject a fake `PublishAPI` now instead of having to use jest to mock the import.
  * We still deep require the `parsers` to pass in the tests, but this could be mocked as well since I feel like the purpose in these RPC tests is not to test the parsers. (Ideally the parsers would be tested separately.) This would require rewriting larger parts of these tests however, so I left it out. cc @MayaUribe, would love to know what your thoughts are on that!

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/server/rpc/index.js [hunk 1]
```diff
@@ -1,173 +1,32 @@
...
-  removeUserTag,
-  getShopGridPostsMethod,
-  updateShopGridPostLink,
-  updateCustomLinksMethod,
-  updateSingleCustomLinkMethod,
-  deleteCustomLinkMethod,
-  createCampaign,
-  updateCampaign,
-  getCampaign,
+async function makeRPCHandler() {
```
Comment (MayaUribe) [inline packages/server/rpc/index.js]: This is great @hamstu! ❤️

File: packages/server/rpc/campaigns/create/index.test.js [hunk 2]
```diff
@@ -29,9 +24,20 @@ const CREATE_CAMPAIGN_RESPONSE = {
   success: true,
 };
 
+const PublishAPI = {
```
Comment (MayaUribe) [inline packages/server/rpc/campaigns/create/index.test.js]: 🙌

**Comment by hamstu** [comment]
Thanks so much for the speedy review @MayaUribe! 💯

File: packages/server/rpc/index.js (hunk 1 above)
Comment (hamstu) [reply packages/server/rpc/index.js]: Thanks! It was starting to feel a little crazy how big that file was growing. 😅

**Pull request merged**